terratest abstraction (#514)
k8gb-io/k8gb: terratest/README.md

# K8GB terratests

> Terratests are another layer in k8gb testing. They test the state of k8b on tuple of clusters and examine the behavior 

Reviewer: `s/k8b/k8gb`

> for individual usecases. They replace manual tests; install GSLB in a separate namespace and check if everything 

Reviewer: ```suggestion
for individual use cases. They replace manual tests, install GSLB in a separate namespace and check if everything 
```

behaves as expected.

> Terratest is basically a separate project inside k8gb. The tests are running as Gihub actions when interacting 

Reviewer: ```suggestion
Terratest is basically a separate project inside k8gb. The tests are running as GitHub Actions when interacting 
```

with remote repo, or you can run them locally with `make terratest`. To run it locally, you must have k3d clusters 
with the k8gb operator installed. Read [Quick Start](../README.md#quick-start) if you have not already done so.

Terratests consist of three directories:

- `/test/` contains the individual tests. Each test usually creates its own namespace where it installs its own 
  version of GSLB. The test then runs against such instance. For optimization reasons, all tests are running 
  in parallel. Keep this on mind!
- `/examples/` contains yaml configurations (GSLB, ingresses) for individual tests.
- `/utils/` contains a common framework that makes writing tests easier.

## Terratest Common Framework
First of all, nothing forces you to use the framework. There are certainly many cases where it is better not to use 
the framework and instead interact with the cluster directly using terratest (which is itself a powerful framework).

> On the other hand, the ability to quickly spin-up clusters with test applications and then easily read ip addresses 

Reviewer: ```suggestion
On the other hand, the ability to quickly spin up clusters with test applications and then easily read ip addresses 
```
